### PR TITLE
Add warning messages for bad keybindings

### LINKED ISSUE #3522: Display a warning if a keybinding didn't have a required arg
From work in #3391. Some `ShortcutActions` have _required_ arguments. If a binding has a required arg without a reasonable default value, and the user fails to provide that arg, then we should display a warning dialog.

Case in point, `ResizePane` and `MoveFocus`. 

related: #3475

### LINKED ISSUE #4239: Display a warning when a user tries to bind an action to multiple keys in a single chord
After upgrading to WT 0.8, I added this to my `profile.json`:
```
"keybindings":
    [
        // "index": 1 ==> "Ubuntu"
        { "keys": ["ctrl+2+plus"],
          "command": {"action": "splitPane", "split": "vertical", "index": 1}
        },
        { "keys": ["ctrl+2+-"],
          "command": {"action": "splitPane", "split": "horizontal", "index": 1}
        }//,
        // "index": 2 ==> "CMD"
        // { "keys": ["ctrl+3+plus"],
        //   "command": {"action": "splitPane", "split": "vertical", "index": 2}
        // },
        // { "keys": ["ctrl+3+-"],
        //   "command": {"action": "splitPane", "split": "horizontal", "index": 2}
        // }
    ]
```
and it works as expected: profile `index 0` is MSYS2 and `index 1` is Ubuntu, so `ctrl+2+...` can split current profile calling Ubuntu.

If I uncomment the other lines (`index 2` for CMD), both `ctrl+2+...` and `ctrl+3+...` open CMD profile. In short adding similar lines for other profile, always open the last profile and not what they should open...

I wonder what is wrong with those lines.. maybe i have misunderstood? Thanks...

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Adds warning messages for a pair of keybindings-related scenarios. This covers the following two bugs:
* #4239 - If the user has supplied more than one key chord in their `"keys"` array.
* #3522 - If a keybinding has a _required_ argument, then we'll display a message to the user
  - currently, the only required parameter is the `direction` parameter for both `resizePane` and `moveFocus`

## References

When we get to #1334, we'll want to remove the `TooManyKeysForChord` warning.

## PR Checklist
* [x] Closes #4239
* [x] Closes #3522
* [x] I work here
* [x] Tests added/passed
* [n/a] Requires documentation to be updated

image

## Validation Steps Performed
Tested manually, added tests.

### CODE REVIEW

File: src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp [hunk 1]
```diff
@@ -146,6 +146,10 @@ static const std::map<std::string_view, ShortcutAction, std::less<>> commandName
     { FindKey, ShortcutAction::Find },
 };
 
+// using ParseActionFunction = std::function<IActionArgs(const Json::Value
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp]: stray comment

File: src/cascadia/TerminalApp/ActionArgs.h [hunk 2]
```diff
@@ -221,15 +224,22 @@ namespace winrt::TerminalApp::implementation
...
             // LOAD BEARING: Not using make_self here _will_ break you in the future!
             auto args = winrt::make_self<ResizePaneArgs>();
             if (auto directionString{ json[JsonKey(DirectionKey)] })
             {
                 args->_Direction = ParseDirection(directionString.asString());
             }
-            return *args;
+            if (args->_Direction == TerminalApp::Direction::None)
+            {
+                return { nullptr, { ::TerminalApp::SettingsLoadWarnings::MissingRequiredParameter } };
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/ActionArgs.h]: Just curious, you can still return `{ *args, { ... } }` right?

File: src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp [hunk 3]
```diff
@@ -443,6 +453,15 @@ void winrt::TerminalApp::implementation::AppKeyBindings::LayerJson(const Json::V
         {
             const auto validString = keys.isString();
             const auto validArray = keys.isArray() && keys.size() == 1;
+
+            // microsoft/terminal#4239 - If the user provided more than one key
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp]:
```suggestion
            // GH#4239 - If the user provided more than one key
```
to be consistent

**Comment by miniksa** [comment]
Was going to approve, but I think Carlos is on top of this one as soon as you finish his comments.

**Comment by DHowett-MSFT** [comment]
We can't really add info about the failing args without tearing up our existing system.

File: src/cascadia/TerminalApp/ActionArgs.h (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/ActionArgs.h]: I mean yea you could - but then the (theoretically) malformed args would still be added to the keybinding. Returning null will prevent the key fro getting bound.

**Comment by DHowett-MSFT** [comment]
@carlos-zamora no concerns left, I presume?

**Pull request merged**